### PR TITLE
upgrade `Quamotion.RemoteViewing` to 1.1.211 to work with `RealVNC Viewer`, Add `password` parameter to `StartWithHeadlessVncPlatform`

### PR DESCRIPTION
**Pull request by dameng324**
## What does the pull request do?

upgrade Quamotion.RemoteViewing to latest version to work with `RealVNC Viewer` discussed in https://github.com/AvaloniaUI/Avalonia/discussions/15336#discussioncomment-9128576

## What is the current behavior?

1. RealVNC Viewer can not connect to a Avalonia VNC Server. Only TightVNC Viewer can connect it. TightVNC only have Windows client, RealVNC Viewer has many platform supported.

2. There is no way to set password to protect application.

## What is the updated/expected behavior with this PR?

1. RealVNC Viewer can connect to a Avalonia VNC Server and work as expected.
2. User can set password to protect application.

## How was the solution implemented (if it's not obvious)?

1. Quamotion.RemoteViewing's latest version support RealVNC Viewer, so first, upgrade it  to the latest version.
2. implement the new method of the interface(`IVncFramebufferSource`) we used.
3. add the `Logger` parameter, otherwise It will load unhandled exception when a connection is closed in the `Quamotion.RemoteViewing` library internal.
4. add the `password` parameter to allow user set the vnc connection password.

## Requirement
Windows is out of box to use it, but linux/MacOS must install `jpeg-turbo` library before use it.

ubuntu:`apt install libturbojpeg`
mac: `brew install jpeg-turbo`

## Checklist

- [ ] Added unit tests (if possible) 
   I do not know how to unit test it. but I have tested in my local environment.
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation
   There is no document releate to Avalonia.Headless.VNC, so I do not know where to add it.

## Breaking changes
No break changes here.

## Obsoletions / Deprecations
There is no obsoletions or deprecations.

## Fixed issues
No issue relate to this,But a discuss: https://github.com/AvaloniaUI/Avalonia/discussions/15336

### CODE REVIEW

File: src/Headless/Avalonia.Headless.Vnc/HeadlessVncPlatformExtensions.cs [hunk 1]
```diff
@@ -16,10 +19,14 @@ public static class HeadlessVncPlatformExtensions
         public static int StartWithHeadlessVncPlatform(
             this AppBuilder builder,
             string host, int port,
-            string[] args, ShutdownMode shutdownMode = ShutdownMode.OnLastWindowClose)
+            string[] args,
+            ShutdownMode shutdownMode = ShutdownMode.OnLastWindowClose,
+            string? password = null,
+            ILogger? logger = null)
```
Comment (maxkatz6) [inline src/Headless/Avalonia.Headless.Vnc/HeadlessVncPlatformExtensions.cs:25]: ILogger here is a leaking abstraction. Avalonia itself doesn't reference `Microsoft.Extensions.Logging` in any way, and instead we use `Avalonia.Logging.Logger.TryGet()`. You possibly can write an internal ILogger wrapper here. 
And call `Logger.TryGet(LogLevel, LogArea.VncPlatform)?.Log()` from the wrapper.

**Comment by maxkatz6** [comment]
> As latest version of Quamotion.RemoteViewing does not support netstander2.0, so I removed it from targetframeworks.

It might be a problem. Removing TFM is a big breaking change.
We potentially can backport it to 11.1 release until it's too late. 

cc @kekekeks for an opinion.

**Comment by dameng324** [comment]
It's possable to make it compatible with `netstandard2.0`.

`Quamotion.RemoteViewing` depends on `Quamotion.TurboJpegWrapper` package which only support `netstandard2.1`, but I clone the code, and add reference to `System.Memory` to it when `netstandard2.0`. like this:

Quamotion.TurboJpegWrapper.csproj
```xml
<ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
    <PackageReference Include="System.Memory" Version="4.5.4" />
</ItemGroup>
```

Then it will compatible with netstandard2.0 instead of netstandard2.1, and `Quamotion.RemoteViewing` will compatible with `netstandard2.0` too.

The problem is it need to modify this two package's code and use it. `Quamotion.TurboJpegWrapper` has been archived and `Quamotion.RemoteViewing` repo's last commit is on 2 years ago. 

I can upload a new package to solve this problem, but really don't want do it. What will avalonia team do with this situation? @maxkatz6

**Comment by maxkatz6** [comment]
@dameng324 no need to fork/upload these packages. I will discuss with the team, if we can drop netstandard2.0 here.

File: src/Headless/Avalonia.Headless.Vnc/Avalonia.Headless.Vnc.csproj [hunk 2]
```diff
@@ -1,13 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-      <TargetFrameworks>$(AvsCurrentTargetFramework);$(AvsLegacyTargetFrameworks);netstandard2.0</TargetFrameworks>
+      <TargetFrameworks>$(AvsCurrentTargetFramework);$(AvsLegacyTargetFrameworks)</TargetFrameworks>
```
Comment (kekekeks) [inline src/Headless/Avalonia.Headless.Vnc/Avalonia.Headless.Vnc.csproj:4]: Removing a TFM is a minor-ish breaking change and the general policy is  avoid those.

Check if you can get desired results with a version supporting netstandard2.0

Comment (maxkatz6) [reply src/Headless/Avalonia.Headless.Vnc/Avalonia.Headless.Vnc.csproj:4]: Specifically 
https://www.nuget.org/packages/Quamotion.RemoteViewing/1.1.179

Comment (dameng324) [reply src/Headless/Avalonia.Headless.Vnc/Avalonia.Headless.Vnc.csproj:4]: OK，I will try it.

Comment (dameng324) [reply src/Headless/Avalonia.Headless.Vnc/Avalonia.Headless.Vnc.csproj]: 1.1.179 works fine. I will try downgrade to this version.

**Comment by dameng324** [comment]
Downgrade has done. 
Logger issue has been fixed.

But pipeline failed with ValidateApiDiff Errors as I add an optional parameter `password`.
Can you help me to fix this? @maxkatz6

**Comment by maxkatz6** [comment]
@dameng324 you need to add a second method overload to avoid that. Adding new optional parameter is a binary breaking change, unfortunately, so you need a second method.

**Comment by dameng324** [comment]
@maxkatz6  The work is done. All checks have passed. Would you spare some time to take a look?

**Review by maxkatz6 — APPROVED** [review]
Thank you! Looks good.

**Pull request merged**